Lib/fnmatch.py

```diff
@@ -50,15 +49,9 @@ def filter(names, pat):
     result = []
     pat = os.path.normcase(pat)
     match = _compile_pattern(pat)
-    if os.path is posixpath:
-        # normcase on posix is NOP. Optimize it away from the loop.
-        for name in names:
-            if match(name):
-                result.append(name)
-    else:
-        for name in names:
-            if match(os.path.normcase(name)):
-                result.append(name)
+    for name in names:
+        if match(os.path.normcase(name)):
+            result.append(name)
     return result
```

Comment:
This might be a bit faster?

```suggestion
    normcase = os.path.normcase
    match = _compile_pattern(normcase(pat))
    return [name for name in names if match(normcase(name))]
```

Reply:
I was tempted to do that but I never remember whether spawning a loop or a list comprehension are equivalent in terms of memory consumption or not... I *think* so, hence I'll also commit those changes as well but feel free to correct me if needed!